app/test_main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+import pytest
+from unittest import mock
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, expected_result, mocked_hour",
+    [
+        ("", "Not accessible", 5),
+        ("hsgdfjhg", "Not accessible", 2),
+        ("http://www.google.com", "Accessible", 18),
+        ("http://www.google.com", "Not accessible", 4),
+        ("https://www.gogle.com", "Not accessible", 23),
+    ]
+)
+def test_can_access_google_page_get_expected_value(
+        url: str,
+        expected_result: str,
+        mocked_hour: int
+) -> None:
+    with mock.patch("app.main.datetime.datetime") as mocked_datetime:
+        mocked_now = mock.Mock()
+        mocked_now.hour = mocked_hour
+        mocked_datetime.now.return_value = mocked_now
+        assert can_access_google_page(url) == expected_result
+
+
+def test_has_internet_connection_and_can_access_google_page_was_called(
+) -> None:
+    with mock.patch(
+        "app.main.has_internet_connection"
+    ) as mocked_has_connection, mock.patch(
+        "app.main.valid_google_url"
+    ) as mocked_valid_url:
+
+        url = "https://www.google.com"
+        can_access_google_page(url)
+        mocked_has_connection.assert_called_once()
+        mocked_valid_url.assert_called_once()
```

Comment:
Please, create one function that will serve all checks. You should mock the return value for `valid_google_url` and `has_internet_connection`, not their calls or hours. Your input to `@pytest.mark.parametrize` should be something like `"valid_url,has_connection,expected"` and you should test 4 cases: when `valid_google_url` and `has_internet_connection` return either True/True, False/True, True/False, False/False. Besides that, it is preferable to use mocks as decorators to not initialize context manager for a whole function body